### DISCUSSION TITLE
A friendly reminder to the LibreY devs about your instances.json file.

### DISCUSSION THREAD

**Discussion by rjc3rd** [Ahwxorg/LibreY#164] [General]
Your instance of LibreY uses your own instances.json file to fallback on, actually quite often, I think. If an instance on your instances.json file is broken and can't load for some reason, then that particular search will fail.

When I first noticed it was happening, I noticed that it was always the same instance that failed to get a result. So I went to the instances.json file and found more than a few instances were not working at all. I removed them from my instances list and boom, practically no failed searches.

**Comment by davidovski** [comment]
The intended behaviour should be placing failing instances on cooldown (fallback.php:25-42), that way having broken instances (provided that they are up to date and provide no results when they can't make a query of their own) shouldn't impact the results at the end.

That being said, I think the cooldown on a failing instance might be set too high by default, causing your instance to not have any to fallback to anymore. Try tweaking the request_cooldown setting, this might fix some issues.

I have mentioned this previously, but a better system for selecting fallback instances probably could be put into place, rather than placing instances on cooldown for a set amount of time, it could order instances based on how recently they've been used and worked, that way libreY will be able to make the most of any working instances at any given time and allow for ratelimited instances to have a break before being tried again.

I've been meaning to implement this for months now but I just haven't had the time or motivation, but if you're willing to give it a shot, go for it. I'm happy to discuss any ideas I've had for how this could be implemented
